src/components/ClusterOverview/ClusterOverview.js

```diff
@@ -32,17 +32,18 @@ const ClusterOverview = () => {
     const currentClusterMemoryUsage = await clusterPromql.fetchClusterMemoryUsage();
     const currentTotalDeployments = await clusterPromql.fetchTotalDeployments();
     const currentTotalPods = await clusterPromql.fetchTotalPods();
- 
     setClusterCpuUsage(currentClusterCpuUsage);
     setClusterMemoryUsage(currentClusterMemoryUsage);
     setClusterTotalDeployments(currentTotalDeployments);
     setClusterTotalPods(currentTotalPods);
   };
  
   // fetch data, then fetch again in 30 seconds
-  useEffect(async() => {
+  // when component unmounts, cancel setInterval in a cleanup function
+  useEffect(() => {
     fetchToStore();
-    setInterval(() => fetchToStore(), 30000);
+    const interval = setInterval(() => fetchToStore(), 3000)
```

Comment:
Firing a request every 3 seconds can be quite expensive. I think 30 seconds is good enough.